include/options/options.php

```diff
@@ -0,0 +1,506 @@
+<?php
+/**
+ * @package Polylang
+ */
+
+namespace WP_Syntex\Polylang\Options;
+
+use WP_Error;
+use WP_Syntex\Polylang\Options\Option\Abstract_Option;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class that manages Polylang's options:
+ * - Automatically stores the options into the database on `shutdown` if they have been modified.
+ * - Behaves almost like an array, meaning only values can be get/set (implements `ArrayAccess`).
+ * - Handles `switch_to_blog()`.
+ * - Options are always defined: it is not possible to unset them from the list, they are set to their default value instead.
+ * - If an option is not registered but exists in database, its raw value will be kept and remain untouched.
+ *
+ * @since 3.7
+ *
+ * @implements \ArrayAccess<non-falsy-string, mixed>
+ */
+class Options implements \ArrayAccess {
+	const OPTION_NAME = 'polylang';
+
+	/**
+	 * Polylang's options, by blog ID.
+	 * Raw value if option is not registered yet, `Abstract_Option` instance otherwise.
+	 *
+	 * @var Abstract_Option[][]|mixed[][]
+	 * @phpstan-var array<int, array<non-falsy-string, mixed>>
+	 */
+	private $options = array();
+
+	/**
+	 * Tells if the options have been modified, by blog ID.
+	 *
+	 * @var bool[]
+	 * @phpstan-var array<int, bool>
+	 */
+	private $modified = array();
+
+	/**
+	 * The original blog ID.
+	 *
+	 * @var int
+	 */
+	private $blog_id;
+
+	/**
+	 * The current blog ID.
+	 *
+	 * @var int
+	 */
+	private $current_blog_id;
+
+	/**
+	 * Cached options JSON schema by blog ID.
+	 *
+	 * @var array[]|null
+	 */
+	private $schema;
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.7
+	 */
+	public function __construct() {
+		// Keep track of the blog ID.
+		$this->blog_id = (int) get_current_blog_id();
+
+		// Handle options.
+		$this->init_options_for_blog( $this->blog_id );
+
+		add_action( 'switch_blog', array( $this, 'init_options_for_blog' ), PHP_INT_MIN );
```

Comment:
I don't like blocking things. Also also non-default priorities should be commented.
```suggestion
		add_action( 'switch_blog', array( $this, 'init_options_for_blog' ), -1000 );
```

Reply:
Done in [1230486](https://github.com/polylang/polylang/pull/1451/commits/12304865104f8da0cfc13a6557b4d88449c012c0).